server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java

```diff
@@ -1896,7 +1895,6 @@ public void testForceMergeWithSoftDeletesRetentionAndRecoverySource() throws Exc
                 numSegments = searcher.getDirectoryReader().leaves().size();
             }
             if (numSegments == 1) {
-                boolean useRecoverySource = randomBoolean() || omitSourceAllTheTime;
```

Comment:
If `numSegments = 1`, we add a new document here, so the flag should be generated randomly.